src/EFCore.MySql/Migrations/Internal/MySqlMigrator.cs

```diff
@@ -0,0 +1,138 @@
+using System;
+using JetBrains.Annotations;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Diagnostics;
+using Microsoft.EntityFrameworkCore.Migrations;
+using Microsoft.EntityFrameworkCore.Migrations.Internal;
+using Microsoft.EntityFrameworkCore.Storage;
+
+namespace Pomelo.EntityFrameworkCore.MySql.Migrations.Internal
+{
+    public class MySqlMigrator : Migrator
+    {
+        public MySqlMigrator(
+            [NotNull] IMigrationsAssembly migrationsAssembly,
+            [NotNull] IHistoryRepository historyRepository,
+            [NotNull] IDatabaseCreator databaseCreator,
+            [NotNull] IMigrationsSqlGenerator migrationsSqlGenerator,
+            [NotNull] IRawSqlCommandBuilder rawSqlCommandBuilder,
+            [NotNull] IMigrationCommandExecutor migrationCommandExecutor,
+            [NotNull] IRelationalConnection connection,
+            [NotNull] ISqlGenerationHelper sqlGenerationHelper,
+            [NotNull] IDiagnosticsLogger<DbLoggerCategory.Migrations> logger,
+            [NotNull] IDatabaseProvider databaseProvider)
+            : base(
+                migrationsAssembly,
+                historyRepository,
+                databaseCreator,
+                migrationsSqlGenerator,
+                rawSqlCommandBuilder,
+                migrationCommandExecutor,
+                connection,
+                sqlGenerationHelper,
+                logger,
+                databaseProvider)
+        {
+        }
+
+        public override string GenerateScript(
+            string fromMigration = null,
+            string toMigration = null,
+            bool idempotent = false)
+        {
+            return Header() + base.GenerateScript(fromMigration, toMigration, idempotent) + Footer();
+        }
```

Comment:
I think the appropriate stored procedures should only be added if they are actually needed by the generated script. Otherwise this could end up as a long list of mostly unused stored procedures in the future.

Reply:
Done